torch/distributed/_shard/checkpoint/metadata.py

```diff
@@ -5,52 +5,39 @@
 import torch
 from torch.distributed._shard.sharded_tensor import (
     ShardedTensor,
-    ShardedTensorMetadata,
-    ShardMetadata,
 )
+from torch.distributed._shard.sharded_tensor.metadata import TensorProperties
 
-TENSOR_TYPE = Union[torch.Tensor, ShardedTensor]
-STATE_DICT_TYPE = Dict[str, Any]
-
-@dataclass
-class ShardStorageMetadata:
-    shard_metadata: ShardMetadata
-    # storage key used for this particular Shard
-    storage_key: str
-
-
-# Metadata for each param.
 @dataclass
-class ShardedTensorStorageMetadata:
-    # Metadata for the sharded tensor itself
-    tensor_metadata: ShardedTensorMetadata
-
-    # Storage info for each Shard. There's no ordering requirement for this list.
-    storage_metadata: List[ShardStorageMetadata]
+class ChunkStorageMetadata:
+    """
+    Each chunk is expected to have the same properties of the TensorStorageMetadata that includes it.
+    """
+    offsets: torch.Size
+    sizes: torch.Size
 
+    # This is reported by the storage layer so it might include serialization overhead
+    size_in_bytes: int
 
 @dataclass
 class TensorStorageMetadata:
-    # Storage key used for this tensor
-    storage_key: str
-
-    # Tensor sizes
+    properties: TensorProperties
     size: torch.Size
+    chunks: List[ChunkStorageMetadata]
 
 @dataclass
 class BytesStorageMetadata:
-    # Storage key used for this tensor
-    storage_key: str
-
-    # serialized payload size
-    length: int
+    size_in_bytes: int
```

Comment:
hmmm what's the point of having `BytesStorageMetadata` as `ChunkStorageMetadata` already have it? Is it because `BytesStorageMetadata` could handle more cases then chunk sharding?

I'm wondering if we can make BytesStorageMetadata be a base class if `size_in_bytes` is common

Reply:
`BytesStorageMetadata` exists to signal that this entry stores non-tensor data.